src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -544,6 +546,245 @@ def _get_connections(self):
+        many_to_many_input_count = []
+        many_to_many_output_count = []
+
+        for name, options, type in zip(connector_names, connector_options, connector_type):
+            variable_list = [var for var in connector_variable if name in var]
+            defined_multi_connection_exists = options is not None and any(
+                key.startswith("number_of_") for key in options.keys()
+            )
+
+            if defined_multi_connection_exists:
+                # This is for the connectors having given input and output numbers from pt file
+                num_connections = [
+                    num
+                    for option, num in zip(options.keys(), options.values())
+                    if option.startswith("number_of_")
+                ]
+
+                # First check if there is any side having multiple connection
+                if any(num > 1 for num in num_connections):
+                    # This is to identify many-to-many component
+                    if all(num > 1 for num in num_connections):
+                        many_to_many_component.append(name)
+                        many_to_many_input_count.append(
+                            int(options.get("number_of_inputs") or options.get("number_of_tanks"))
+                        )
+                        many_to_many_output_count.append(
+                            int(
+                                (
+                                    options.get("number_of_outputs")
+                                    or options.get("number_of_power_sources")
+                                    or options.get("number_of_engines")
+                                )
+                            )
+                        )
+
+                    # This is to identify many-to-one component
+                    elif (options.get("number_of_inputs") or options.get("number_of_tanks")) > 1:
+                        many_to_one_component.append(name)
+                        many_to_one_input_count.append(
+                            int(options.get("number_of_inputs") or options.get("number_of_tanks"))
+                        )
+                    # This is to identify one-to-many component
+                    else:
+                        one_to_many_component.append(name)
+                        one_to_many_output_count.append(
+                            int(
+                                options.get("number_of_outputs")
+                                or options.get("number_of_power_sources")
+                                or options.get("number_of_engines")
+                            )
+                        )
+                else:
+                    one_to_one_component.append(name)
+
+            else:
+                if type == "DC_splitter" or type == "planetary_gear":
+                    many_to_one_component.append(name)
+                    many_to_one_input_count.append(2)
+
+                elif type == "gearbox":
```

Comment:
Don't gearbox always have 2 outputs ? Speed reducers on the other hand are one to one